### ISSUE TITLE
[New metric] Deviance scores

### ISSUE THREAD

**Issue by SkafteNicki** [PyTorchLightning/metrics#73]
## 🚀 Feature

Add deviance scores:
* Poisson deviance https://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_poisson_deviance.html#sklearn.metrics.mean_poisson_deviance
* Gamma deviance https://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_gamma_deviance.html#sklearn.metrics.mean_gamma_deviance
* Tweedie deviance https://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_tweedie_deviance.html#sklearn.metrics.mean_tweedie_deviance

Use in regression tasks and for goodness-of-fit testing.

**Comment by hassiahk** [comment]
Hi @SkafteNicki, can I also work on this simultaneously with `KLD`?

**Comment by SkafteNicki** [comment]
@hassiahk yes, I will assign you :]

**Comment by hassiahk** [comment]
Hello @SkafteNicki, should we have three different metrics for this?

`Tweedie deviance with power`
`Poisson deviance - Tweedie deviance with power=1`
`Gamma deviance - Tweedie deviance with power=2`

**Comment by SkafteNicki** [comment]
> 
> 
> Hello @SkafteNicki, should we have three different metrics for this?
> 
> `Tweedie deviance with power`
> `Poisson deviance - Tweedie deviance with power=1`
> `Gamma deviance - Tweedie deviance with power=2`

I think in this case, it is better to just have a single metric called `Deviance`  (or `TweedieDeviance`) and then make it clear in the docstring what `power=1`  and `power=2` corresponds to

**Comment by hassiahk** [comment]
> I think in this case, it is better to just have a single metric called `Deviance` (or `TweedieDeviance`) and then make it clear in the docstring what `power=1` and `power=2` corresponds to

Yes, that makes sense :)

**Comment by Borda** [comment]
@hassiahk @SkafteNicki how is it going here? :rabbit:

**Comment by hassiahk** [comment]
> @hassiahk @SkafteNicki how is it going here? 🐰

I was working on it and then got busy with something else. I will raise a draft PR by tomorrow.

**Comment by hassiahk** [comment]
@Borda I will be drafting a PR for this by this weekend. Sorry for the delay.

**Comment by SkafteNicki** [comment]
@hassiahk sounds great :]
